flink-streaming-connectors/flink-connector-redis/src/main/java/org/apache/flink/streaming/connectors/redis/RedisSink.java

```diff
@@ -0,0 +1,235 @@
+		Preconditions.checkNotNull(redisSinkMapper, "Redis Mapper can not be null");
+
+		this.redisSinkMapper = redisSinkMapper;
+		this.jedisPoolConfig = jedisPoolConfig;
+		RedisDataTypeDescription dataTypeDescription = redisSinkMapper.getDataTypeDescription();
+		this.redisDataType = dataTypeDescription.getDataType();
+		this.additionalKey = dataTypeDescription.getAdditionalKey();
+
+	}
+
+	/**
+	 * Creates a new RedisSink that connects to the Redis Sentinels.
+	 *
+	 * @param jedisSentinelConfig The configuration of {@link org.apache.flink.streaming.connectors.redis.common.config.JedisSentinelConfig}
+	 * @param redisSinkMapper This used for generate redis command and key value from incoming elements
+	 */
+	public RedisSink(JedisSentinelConfig jedisSentinelConfig, RedisMapper<IN> redisSinkMapper) {
+		Preconditions.checkNotNull(jedisSentinelConfig, "Redis Sentinel connection pool config should not be Null");
+		Preconditions.checkNotNull(redisSinkMapper, "Redis Mapper can not be null");
+
+		this.redisSinkMapper = redisSinkMapper;
+		this.jedisSentinelConfig = jedisSentinelConfig;
+		RedisDataTypeDescription dataTypeDescription = redisSinkMapper.getDataTypeDescription();
+		this.redisDataType = dataTypeDescription.getDataType();
+		this.additionalKey = dataTypeDescription.getAdditionalKey();
+
+	}
+
+	/**
+	 * Creates a new RedisSink that connects to the Redis Cluster.
+	 *
+	 * @param jedisClusterConfig The configuration of {@link org.apache.flink.streaming.connectors.redis.common.config.JedisClusterConfig}
+	 * @param redisSinkMapper This used for generate redis command and key value from incoming elements
+	 */
+	public RedisSink(JedisClusterConfig jedisClusterConfig, RedisMapper<IN> redisSinkMapper) {
+		Preconditions.checkNotNull(jedisClusterConfig, "Redis cluster config should not be Null");
+		Preconditions.checkNotNull(redisSinkMapper, "Redis Mapper can not be null");
+
+		this.redisSinkMapper = redisSinkMapper;
+		this.jedisClusterConfig = jedisClusterConfig;
+		RedisDataTypeDescription dataTypeDescription = redisSinkMapper.getDataTypeDescription();
+		this.redisDataType = dataTypeDescription.getDataType();
+		this.additionalKey = dataTypeDescription.getAdditionalKey();
+
+	}
+
+	/**
+	 * Called when new data arrives to the sink, and forwards it to Redis channel.
+	 * <p> If redis data type is {@link RedisDataType#HASH} Redis HSET command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#LIST} Redis RPUSH command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#SET} Redis SADD command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#PUBSUB} Redis PUBLISH command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#STRING} Redis SET command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#HYPER_LOG_LOG} Redis PFADD command will be applied.
+	 * <p> If redis data type is {@link RedisDataType#SORTED_SET} Redis ZADD command will be applied.
```

Comment:
This is somewhat redundant. Maybe better: "Depending on the specified Redis data type (see {@link RedisDataType}), a different Redis command will be applied. Available commands are HSET, RPUSH, SADD, PUBLISH, SET, PFADD, and ZADD."